### ISSUE TITLE
don't package all of texlive in haikuports

### ISSUE THREAD

**Issue by jmairboeck** [haikuports/haikuports#6771]
Now we have a texlive package containing everything that texlive offers, but marked as untested because it is too big to host by Haiku. What should be done instead is to only provide packages for the most important stuff and packages that others depend on (e.g. dblatex, see #6078). The rest could be installed by users directly from CTAN if needed, e.g. via tlmgr that is provided by texlive.

**Comment by pulkomandy** [comment]
Fore reference: it needs 4GB. I don't think that's unmanageable by today's standards?

**Comment by Begasus** [comment]
If you count both 32bit and 64bit that adds up to 8GB?

**Comment by jmairboeck** [comment]
> If you count both 32bit and 64bit that adds up to 8GB?

If the package building and hosting is intelligent enough, then not, because these are "any" packages suitable for any architecture.

**Comment by Begasus** [comment]
Right, forgot about that, it's not my day today ;)

**Comment by Begasus** [comment]
Got this earlier when trying to run TeXstudio on a simple tex file:
```
! I can't find file `loadhyph-af.tex'. <to be read again> \relax l.192 \repeat (Press Enter to retry, or Control-D to exit) Please type another input file name: ! Emergency stop. <to be read again> \relax l.192 \repeat ! ==> Fatal error occurred, no output PDF file produced! Transcript written on pdflatex.log.
```
Looking into the source that file is located in `texmf-dist/tex/generic/hypth-utf8` ? (can't seem to locate it in any of the packages?)

@jmairboeck any idea?

**Comment by jmairboeck** [comment]
Without actually having it checked: you can search for particular files using `kpsewhich`, so `kpsewhich loadhyph-af.tex` should find it if it is installed. I noticed that the tex engines require probably all of them because they are configured to have all languages available. I would consider that a packaging bug, but have not investigated further how to fix that. That is also the reason why the lilypond recipe depends on (nearly?) all the `texlive_lang*` packages:

https://github.com/haikuports/haikuports/blob/bd1623d73b487df59e6506c018ee0fb210f9d00d/media-sound/lilypond/lilypond-2.22.2.recipe#L70-L85

The way how I determined that list was using `catattr SYS:PACKAGE $(kpsewhich filename)` (having `texlive_full` installed) for all files that reported missing, iteratively.

In that concrete case, `af` stands probably for Afrikaans, so I would suggest installing `texlive_langother`. But I wouldn't be surprised if you indeed need the full list as in the lilypond recipe.

**Comment by Begasus** [comment]
Think I overlooked the content of texlive_langother when I was looking for that file :) now it's listed with kpsewhich :+1: 
Isn't the list somewhere contained in a fmt file?

Will report back later.

**Comment by jmairboeck** [comment]
I got a personal email from someone (@gkourgkoutas) yesterday asking how to install texlive on Haiku. This is not a very pleasant situation for users, having to build texlive for themselves because it is disabled.

**Comment by Brunobasta** [comment]
I totally agree on the fact that it is not a pleasant situation for users to build texlive.

Many German students for example use TeXlive for study math/physics (like my son).
I think it is a must have to support the students at universities to make Haiku more popular!
If it is that big, put it on another repository? Optional package for education/study, or jut to besly for example?

We have full guide on how to install the latest texlive onto Haiku. (Thanks to okabe)!
It is on our community forum:
https://discuss.haiku-os.org/t/texlive-on-x86-64/10548/37

**Comment by Begasus** [comment]
The situation is as it is atm, once the main contributors agree on this, this could be resolved at haikuports, until then it's up to the users to build this them selves ... The post in the forum is nice for those not familiar with haikuporter, for those that are, it should be clear on how to build and install the packages

**Comment by jmairboeck** [comment]
The even better solution would be if texlive would be patched so that the `tlmgr` (TeX Live Manager) could be used to install packages to non-packaged, similar to how pip works for python. Then we could remove most of the content from the existing texlive package and only package the basics and things needed by other packages as individual recipes.

I am currently not really motivated to try that though. This seems to be non-trivial.

**Comment by pulkomandy** [comment]
What is there to agree? Wasn't the pacbaging already reworked to build several smaller packages instead of a single very large one? That's how it's done in Debian for example and I think it's fine.

**Comment by Begasus** [comment]
Yes, and working fine from what I've been able to test, at the time back then it wasn't enabled because of the size of the package(s), if space on buildmaster isn't an issue anymore I'm all in to enable the recipes for this.

**Comment by Begasus** [comment]
Maybe the main people to ask if this can be enabled would be @mmlr or @kallisti5 (don't shoot if I'm wrong here :) )

**Comment by korli** [comment]
Are other OSs also producing 4gb packages for texlive?

**Comment by jmairboeck** [comment]
most Linux distributions package all of texlive, in varying granularity. IIRC Debian does something similar to Haiku (packages per texlive "collection"). In OpenSUSE, every Tex package is packaged individually, and there are a few thousand of them.

**Comment by gkourgkoutas** [comment]
> Are other OSs also producing 4gb packages for texlive?

The texlive-full package under Ubuntu is ~6GB IIRC.

**Comment by Begasus** [comment]
Did a file listing yesterday, to have a working texlive environment I had to install about 593.37 MiB (22 packages), the full suite is about 4.38 GiB (125 packages) on 32bit. So on the hosting side it would mean about 9 GiB to host the files for both 32bit and 64bit.
If we would only enable it for 64bit (I think most users will use it there), it will reduce it on the hosting side to the half of that.
Repology reports texlive for all main OS's: https://repology.org/project/texlive/versions
Fedora also produces a extended list of subpackages: https://packages.fedoraproject.org/pkgs/texlive/

**Comment by jmairboeck** [comment]
Only enabling it for a particular architecture is not possible without changing the ARCHITECTURE (wrongly) because texlive is an "any" arch package, and it makes no sense.

What should be done is building the package only once and re-using it for all architectures. That is what I meant with "deduplication" on IRC.

**Comment by Begasus** [comment]
My bad, was only thinking out loud ;)

**Comment by korli** [comment]
> Did a file listing yesterday, to have a working texlive environment I had to install about 593.37 MiB (22 packages), the full suite is about 4.38 GiB (125 packages) on 32bit. 

So why was a 4gb package mentioned in this thread?

**Comment by jmairboeck** [comment]
It was a single 4 GiB package before splitting it up in #6229.

**Comment by Begasus** [comment]
The main question atm is if we could agree to host the suite on the server, on the account on what the user needs is a different thing and doesn't need an install for 4GiB

**Comment by pulkomandy** [comment]
4GB on the server is no problem on the server, and we have wasted enough time talking about this over and over. Why can't we just simply enable this and move on? Is there a pull request for enabling it that we can just merge?

**Comment by Brunobasta** [comment]
Wow that is quite a bunch of files...
But 4GB is not much if it works out of the box! Some games using much more these days, which we should not focus on!

Anyway if it is needed to work reliable with TeX so it is a must have! Otherwise students won't use it.

Begasus which E-Mail do you use for testing my TeX files?

**Comment by Begasus** [comment]
ATM I'm checking a build without the doc packages (those count up to 2.58 GiB in total), not ideal but would save at least  about 5 GiB on hosting ...

**Comment by korli** [comment]
can this be closed?

**Comment by pulkomandy** [comment]
Yes, it is enabled now and nothing collapsed.

**Comment by jmairboeck** [comment]
Well, technically it was not done as described here, because that would mean replacing the `texlive` recipe with multiple individual recipes for the TeX packages we want to package (in the `dev-texlive` category) and not using install-tl to generate the config files but creating them ourselves (which would fix #7828 if done correctly).

But I doubt that this will be done soon (I will probably not do it), so I guess it is okay closing this.
